sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -392,4 +392,236 @@ case class GpuArrayMax(child: Expression) extends GpuBaseArrayAgg with ImplicitC
   override def prettyName: String = "array_max"
 
   override protected def agg: GroupByAggregation = GroupByAggregation.max()
-}
\ No newline at end of file
+}
+
+class GpuSequenceMeta(
+    expr: Sequence,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends ExprMeta[Sequence](expr, conf, parent, rule) {
+
+  override def convertToGpu(): GpuExpression = {
+    if (expr.stepOpt.isDefined) {
+      val Seq(start, stop, step) = childExprs.map(_.convertToGpu())
+      GpuSequenceWithStep(start, stop, step, expr.timeZoneId)
+    } else {
+      val Seq(start, stop) = childExprs.map(_.convertToGpu())
+      GpuSequence(start, stop, expr.timeZoneId)
+    }
+  }
+}
+
+/** GpuSequence without step */
+case class GpuSequence(start: Expression, stop: Expression, timeZoneId: Option[String] = None)
+    extends GpuBinaryExpression with TimeZoneAwareExpression {
+
+  override def left: Expression = start
+
+  override def right: Expression = stop
+
+  override def dataType: DataType = ArrayType(start.dataType, containsNull = false)
+
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
+    copy(timeZoneId = Some(timeZoneId))
+
+  /** Calculate the size between start and stop, both inclusive
+   * |stop - start| + 1 */
+
+  /**
+   * Calculate the size between start and stop both inclusive and the step
+   * @param start
+   * @param stop
+   * @return (size, step)
+   */
+  private def calculateSizeAndStep(start: BinaryOperable, stop: BinaryOperable):
+      Seq[ColumnVector] = {
+    withResource(stop.sub(start)) { difference =>
+      withResource(Scalar.fromInt(1)) { scalarOne =>
```

Comment:
Should be `GpuScalar(1, dataType)` or similar, so we can support various types not just integer.

Reply:
Done. Thx